pkg/operation/botanist/controlplane/kubecontrollermanager/kube_controller_manager.go

```diff
@@ -0,0 +1,529 @@
+	}
+
+	var (
+		vpa        = k.emptyVPA()
+		service    = k.emptyService()
+		deployment = k.emptyDeployment()
+
+		labels = map[string]string{
+			v1beta1constants.LabelApp:  v1beta1constants.LabelKubernetes,
+			v1beta1constants.LabelRole: labelRole,
+		}
+		labelsWithControlPlaneRole = utils.MergeStringMaps(labels, map[string]string{
+			v1beta1constants.DeprecatedGardenRole: v1beta1constants.GardenRoleControlPlane,
+		})
+		vpaUpdateMode = autoscalingv1beta2.UpdateModeAuto
+
+		port           = k.computeServerPort()
+		probeURIScheme = k.computeServerURIScheme()
+		command        = k.computeCommand(port)
+	)
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, vpa, func() error {
+		vpa.Spec.TargetRef = &autoscalingv1.CrossVersionObjectReference{
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+			Kind:       "Deployment",
+			Name:       v1beta1constants.DeploymentNameKubeControllerManager,
+		}
+		vpa.Spec.UpdatePolicy = &autoscalingv1beta2.PodUpdatePolicy{
+			UpdateMode: &vpaUpdateMode,
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, service, func() error {
+		service.Labels = labels
+		service.Spec.Selector = labels
+		service.Spec.Type = corev1.ServiceTypeClusterIP
+		service.Spec.Ports = kutil.ReconcileServicePorts(service.Spec.Ports, []corev1.ServicePort{
+			{
+				Name:     portNameMetrics,
+				Protocol: corev1.ProtocolTCP,
+				Port:     port,
+			},
+		})
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutil.CreateOrUpdate(ctx, k.client, deployment, func() error {
+		deployment.Labels = labelsWithControlPlaneRole
+		deployment.Spec.Replicas = &k.replicas
+		deployment.Spec.RevisionHistoryLimit = pointer.Int32Ptr(0)
+		deployment.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
+		deployment.Spec.Template = corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Annotations: map[string]string{
+					"checksum/secret-" + k.secrets.Ca.Name:                k.secrets.Ca.Checksum,
+					"checksum/secret-" + k.secrets.ServiceAccountKey.Name: k.secrets.ServiceAccountKey.Checksum,
+					"checksum/secret-" + k.secrets.Kubeconfig.Name:        k.secrets.Kubeconfig.Checksum,
+					"checksum/secret-" + k.secrets.Server.Name:            k.secrets.Server.Checksum,
+				},
+				Labels: utils.MergeStringMaps(labelsWithControlPlaneRole, map[string]string{
+					v1beta1constants.LabelPodMaintenanceRestart:         "true",
+					v1beta1constants.LabelNetworkPolicyToDNS:            v1beta1constants.LabelNetworkPolicyAllowed,
+					v1beta1constants.LabelNetworkPolicyToShootAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+					v1beta1constants.LabelNetworkPolicyFromPrometheus:   v1beta1constants.LabelNetworkPolicyAllowed,
+				}),
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            v1beta1constants.DeploymentNameKubeControllerManager,
```

Comment:
please define a constant for the containerName, see https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/controlplane/kubescheduler/kube_scheduler.go#L61